src/components/Leaderboard/Leaderboard.tsx

```diff
@@ -0,0 +1,136 @@
+
+export const LEADERBOARD_LENGTH = 25;
+
+const isValidPage = (pageNum: number, totalPages: number) => {
+  return !isNaN(pageNum) && pageNum >= 1 && pageNum <= totalPages;
+};
+
+export const Leaderboard = ({ page: defaultPage }: { page: number }) => {
+  const { account } = useAccount();
+
+  const { t } = useTranslation();
+
+  const {
+    data: leaderboardData,
+    meta,
+  }: { data: LeaderboardEntryData[]; meta: LeaderboardMeta } =
+    useLeaderboardList(defaultPage, LEADERBOARD_LENGTH);
+  const { data: leaderboardUserData }: { data: LeaderboardEntryData } =
+    useLeaderboardUser(account?.address);
+
+  const currentPage = useMemo(() => {
+    return isValidPage(defaultPage, meta?.pagination?.pagesLength)
+      ? defaultPage
+      : 1;
+  }, [defaultPage, meta?.pagination.pagesLength]);
+
+  // const date = new Date().toLocaleDateString('en-GB', {
+  //   day: 'numeric',
+  //   month: 'long',
+  //   year: 'numeric',
+  // });
+
+  return (
+    <PageContainer>
+      <LeaderboardContainer>
+        <LeaderboardHeader>
+          <LeaderboardTitleBox>
+            <Typography variant="headerMedium">
+              {t('leaderboard.title')}
+            </Typography>
+            <LeaderboardUpdateDateBox>
+              <IconHeader
+                tooltipKey={t('leaderboard.description')}
+                title={`Updated: ${t('format.date', { value: new Date() })}`}
+              />
+            </LeaderboardUpdateDateBox>
+          </LeaderboardTitleBox>
+        </LeaderboardHeader>
+        <LeaderboardUserEntry />
+        <LeaderboardEntryStack direction={'column'}>
+          {!leaderboardData?.length
+            ? Array.from({ length: LEADERBOARD_LENGTH }).map((_, index) => (
+                <>
+                  <LeaderboardEntrySkeleton
+                    key={`leaderboard-entry-${index}-skeleton`}
+                    isUserPosition={false}
+                  />
+                  {index !== LEADERBOARD_LENGTH - 1 && (
+                    <LeaderboardEntryDivider
+                      key={`leaderboard-entry-${index}-skeleton-divider`}
+                    />
+                  )}
+                </>
+              ))
+            : leaderboardData?.map(
+                (entry: LeaderboardEntryData, index: number) => {
+                  const isUserPosition =
+                    +parseInt(entry.position) ===
+                    +parseInt(leaderboardUserData?.position);
+
+                  return (
+                    <>
+                      <LeaderboardEntry
+                        key={`leaderboard-entry-${index}`}
+                        isUserPosition={isUserPosition}
+                        walletAddress={entry.walletAddress}
+                        position={
+                          typeof entry.position === 'string'
+                            ? parseInt(entry.position)
+                            : entry.position
+                        }
+                        points={
```

Comment:
You should be able to pass entry.points and entry.points directly

Reply:
yes